### PR TITLE
Introduce Helm good practices in Rafter chart

### PR DESCRIPTION
**Pull request by aerfio**
**Description**

Changes proposed in this pull request:

- as the title says

**Related issue(s)**

See https://github.com/kyma-project/kyma/issues/6509
See https://github.com/kyma-project/kyma/issues/6632

### CODE REVIEW

File: resources/rafter/values.yaml [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
   istio:
     gateway:
       name: kyma-gateway
 
 tests:
   enabled: true
   image:
-    dir: develop/
-    version: e1e8482a
+    repository: "eu.gcr.io/kyma-project"
```
Comment (magicmatatjahu) [inline resources/rafter/values.yaml]: "eu.gcr.io/kyma-project/rafter-test"

File: resources/rafter/templates/tests/test.yaml [hunk 2]
```diff
@@ -1,49 +1,37 @@
...
-      serviceAccountName:  {{ .Chart.Name }}-tests
-      restartPolicy: Never
+      serviceAccountName:  {{ include "rafterTests.fullname" . }}
+      restartPolicy: {{ .Values.tests.restartPolicy }}
       containers:
-        - name: tests
-          image: {{ .Values.global.containerRegistry.path }}/{{ .Values.tests.image.dir }}rafter-test:{{ .Values.tests.image.version }}
-          imagePullPolicy: IfNotPresent
+        - name: {{ include "rafterTests.name" . }}
+          image: "{{ .Values.tests.image.repository }}/rafter-test:{{ .Values.tests.image.tag }}"
```
Comment (magicmatatjahu) [inline resources/rafter/templates/tests/test.yaml]: `"{{ .Values.tests.image.repository }}:{{ .Values.tests.image.tag }}"`

File: resources/rafter/templates/tests/test.yaml [hunk 3]
```diff
@@ -1,49 +1,37 @@
...
+{{ include (print $.Template.BasePath "/_helper_tests.txt") . | indent 14 }}
           env:
             - name: MOCKICE_IMAGE
-              value: "hudymi/mockice:0.1.3"
+              value: "{{ .Values.tests.envs.mockiceImage }}"
             - name: APP_TEST_WAIT_TIMEOUT
-              value: "5m"
+              value: "{{ .Values.tests.envs.waitTimeout }}"
           resources:
             limits:
```
Comment (magicmatatjahu) [inline resources/rafter/templates/tests/test.yaml]: {{ include "rafter.tplValue" ( dict "value" .Values.tests.resources "context" . ) | nindent 12 }}

File: resources/rafter/templates/tests/rbac.yaml [hunk 4]
```diff
@@ -32,19 +30,18 @@ rules:
 kind: ClusterRoleBinding
 apiVersion: rbac.authorization.k8s.io/v1beta1
```
Comment (magicmatatjahu) [inline resources/rafter/templates/tests/rbac.yaml]: apiVersion: rbac.authorization.k8s.io/v1

File: resources/rafter/templates/tests/test.yaml (hunk 3 above)
Comment (aerfio) [reply resources/rafter/templates/tests/test.yaml]: didn't work, nindent wants string, gets `map[string]interface{}`, then crashes

Comment (magicmatatjahu) [reply resources/rafter/templates/tests/test.yaml]: Oh.. sorry. Please add this lines to _helpers.txt

```tpl
{{/*
Renders a value that contains template.
Usage:
{{ include "rafter.tplValue" ( dict "value" .Values.path.to.the.Value "context" $ ) }}
*/}}
{{- define "rafter.tplValue" -}}
    {{- if typeIs "string" .value }}
        {{- tpl .value .context }}
    {{- else }}
        {{- tpl (.value | toYaml) .context }}
    {{- end }}
{{- end -}}
```

File: resources/rafter/templates/tests/test.yaml [hunk 5]
```diff
@@ -1,49 +1,36 @@
...
+              value: "{{ .Values.tests.envs.mockiceImage }}"
             - name: APP_TEST_WAIT_TIMEOUT
-              value: "5m"
+              value: "{{ .Values.tests.envs.waitTimeout }}"
           resources:
-            limits:
-              memory: 128Mi
-{{- end }}
-{{- end }}
+{{ include "rafter.tplValue" ( dict "value" .Values.tests.resources "context" $ ) | indent 14 }}
```
Comment (magicmatatjahu) [inline resources/rafter/templates/tests/test.yaml]: Probably indent `12` not `14`

**Pull request merged**